assets/scripts/streets/StreetName.jsx

```diff
@@ -44,10 +45,61 @@ class StreetName extends React.PureComponent {
     super(props)
 
     this.state = {
-      isHovered: false
+      isHovered: false,
+      modify: false,
+      inpVal: ''
     }
   }
 
+  handleStreetNameChange = () => {
+    let newName = this.state.inpVal
+    if (newName) {
+      this.props.handleStreetNameChange(newName)
+      this.setState({
+        modify: false,
+        inpVal: '',
+        isHovered: false
+      })
+    }
+  }
+
+  handleKeyUp = (evt) => {
+    if (evt.key === 'Enter') {
+      this.handleStreetNameChange()
+    }
+    if (evt.key === 'Escape') {
+      this.setState({
+        modify: false,
+        inpVal: '',
+        isHovered: false
+      })
+    }
+  }
+
+  componentDidMount () {
+    window.addEventListener('mousedown', (event) => {
+      if (!document.getElementById(this.props.id).contains(event.target)) {
+        this.setState({
+          isHovered: false,
+          modify: false,
+          inpVal: ''
+        })
+      }
+    })
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('mousedown', (event) => {
+      if (!document.getElementById(this.props.id).contains(event.target)) {
```

Comment:
When adding and removing event listeners, it is helpful to write a single function rather than repeating them. For instance:

```js
handleClickOutsideComponent = (event) = {
  /* do stuff */
}

componentDidMount () {
  window.addEventListener('mousedown', this.handleClickOutsideComponent)
}

componentWillUnmount () {
  window.removeEventListener('mousedown', this.handleClickOutsideComponent)
}
```